fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/LaoDetailViewModel.java

```diff
@@ -140,6 +143,109 @@ protected void onCleared() {
         disposables.dispose();
     }
 
+    public void endElection(Election election) {
+        Log.d(TAG, "ending election with name : " + election.getName());
+        System.out.println( election.getStartTimestamp());
+        Lao lao = getCurrentLaoValue();
+        if (lao == null) {
+            Log.d(TAG, "failed to retrieve current lao");
+            return;
+        }
+
+        String channel = election.getChannel();
+        String laoId = lao.getChannel().substring(6); // removing /root/ prefix
+        ElectionEnd electionEnd = new ElectionEnd(election.getId(), laoId, election.computerRegisteredVotes());
+
+        try {
+            KeysetHandle publicKeysetHandle = mKeysetManager.getKeysetHandle().getPublicKeysetHandle();
+            String publicKey = Keys.getEncodedKey(publicKeysetHandle);
+            byte[] sender = Base64.getDecoder().decode(publicKey);
```

Comment:
Shouldn't it be Base64URL now?

Reply:
Yes, it was merged with the old version of the master that still had some base64 in the ViewModel, but I will resolve those conflicts and merge with the latest version in an instant :)